### PR TITLE
Migrate java rules to pmd 7.x.x #309:

### PR DESCRIPTION
**Pull request by jborgers**
rule number 31: AvoidLombokAnnotationForNonExistentFields
This one is not difficult

### CODE REVIEW

File: src/main-todo/resources/category/java/common.xml [hunk 1]
```diff
@@ -268,105 +268,6 @@ public class CalendarStuff {
     </rule-->
 
 
-    <rule name="OptimizeMapOrSetForEnum"
```
Comment (stokpop) [inline src/main-todo/resources/category/java/common.xml:271]: OptimizeMapOrSetForEnum is also removed from common.xml: correct?

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/AvoidLombokAnnotationForNonExistentFields.xml [hunk 2]
```diff
@@ -38,10 +37,9 @@ import lombok.Setter;
...
         <expected-problems>5</expected-problems>
-        <expected-linenumbers>4,5,12,19,34</expected-linenumbers>
+        <expected-linenumbers>3,4,11,18,33</expected-linenumbers>
         <code><![CDATA[
-import lombok.Getter;
-import lombok.Setter;
+import lombok.*;
 
 @Getter // bad
 @Setter // bad
```
Comment (stokpop) [inline src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/AvoidLombokAnnotationForNonExistentFields.xml:45]: Weird to have these annotations on Exception anyway?

Wondering: what if there is a class extends and the (only) super class has fields? Might that be valid?

**Review by stokpop — APPROVED** [review]
Looks good, some questions are there not directly related to approving this.

File: src/main-todo/resources/category/java/common.xml (hunk 1 above)
Comment (jborgers) [reply src/main-todo/resources/category/java/common.xml:271]: yes, from the todo-common

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/AvoidLombokAnnotationForNonExistentFields.xml (hunk 2 above)
Comment (jborgers) [reply src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/AvoidLombokAnnotationForNonExistentFields.xml:45]: By my knowledge the annotation should then be at the super class level

Comment (jborgers) [reply src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/AvoidLombokAnnotationForNonExistentFields.xml:45]: indeed weird

**Pull request merged**